src/averageAges.js

```diff
@@ -55,7 +61,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person => {
+      return person.sex === 'm'
+        && people.some(mother => mother.name === person.mother);
+    })
```

Comment:
You can use without `return` 
```suggestion
    ? people.filter(person => person.sex === 'm' && people.some(mother => mother.name === person.mother))
```